src/api/spec/controllers/webui/patchinfo_controller_spec.rb

```diff
@@ -0,0 +1,269 @@
+    end
+
+    it { expect(response).to have_http_status(:success) }
+    it { expect(assigns(:binaries)).to be_a Array }
+    it { expect(assigns(:pkg_names)).to be_empty }
+    it { expect(assigns(:packager)).to eq(user) }
+  end
+
+  describe 'POST #save' do
+    before do
+      login user
+    end
+
+    # This validation is performed at controller level and outside the model
+    context 'with a short summary' do
+      before do
+        post :save, project: user.home_project_name, package: patchinfo_package.name, summary: 'short', description: 'long description ' * 10
+      end
+
+      it { expect(flash[:error]).to eq("|| Summary is too short (should have more than 10 signs)") }
+      it { expect(response).to have_http_status(:success) }
+    end
+
+    # This validation is performed at controller level and outside the model
+    context 'with a short description' do
+      before do
+        post :save, project: user.home_project_name, package: patchinfo_package.name, summary: 'long enough summary is ok', description: 'short'
+      end
+
+      it { expect(flash[:error]).to eq(" || Description is too short (should have more than 50 signs and longer than summary)") }
+      it { expect(response).to have_http_status(:success) }
+    end
+
+    context 'with an unknown issue tracker' do
+      before do
+        post :save, project: user.home_project_name,
+                    package: patchinfo_package.name,
+                    summary: 'long enough summary is ok',
+                    description: 'long enough description is also ok' * 5,
+                    issueid: [769484],
+                    issuetracker: ['NonExistingTracker'],
+                    issuesum: [],
+                    issueurl: ['https://bugzilla.gnome.org/show_bug.cgi?id=769484']
+      end
+
+      it { expect(flash[:error]).to eq("Unknown Issue tracker NonExistingTracker") }
+      it { expect(response).to have_http_status(:success) }
+    end
+
+    context "when the patchinfo's xml is invalid" do
+      before do
+        post :save, project: user.home_project_name,
+                    package: patchinfo_package.name,
+                    summary: 'long enough summary is ok',
+                    description: 'long enough description is also ok' * 5,
+                    issueid: [769484],
+                    issuetracker: ['bgo'],
+                    issuesum: [],
+                    issueurl: ['https://bugzilla.gnome.org/show_bug.cgi?id=769484']
+      end
+
+      it { expect(flash[:error]).to start_with("patchinfo is invalid: ") }
+      it { expect(response).to have_http_status(:success) }
+    end
+
+    context "when the patchinfo's xml is valid" do
+      before do
+        do_proper_post_save
+      end
+
+      it { expect(flash[:notice]).to eq("Successfully edited #{patchinfo_package.name}") }
+      it { expect(response).to redirect_to(action: 'show', project: user.home_project_name, package: patchinfo_package.name) }
```

Comment:
In case of success we probably should also verify the patchinfo got created correctly.